docs/architectural-decision-records/023-seperate-contract-rates-tables-postgres.md

```diff
@@ -0,0 +1,62 @@
+- There are CMS users who additionally need the ability to query and review standalone rates data, regardless if they are shared or not. 
+- Features with relational rates data include Rates Across Submission, Q&A for Rates. 
+- Features with rates queried and handled independently of the contract include Rates Dashboard, rates reporting, rate only submissions.
+
+## Constraints
+- Out of scope: the full database diagram for our new tables or technical discussion of the migration. This is a high-level ADR focused on how we approach health plan data in the application.
+
+## Considered Options
+
+### Option 1 No change to database storage. Continue handling health plan data as binary
+
+This entails continuing our current workarounds in business logic to emulate relational and shared data above the layer of the database. There will be features we cannot adequately deliver that are currently on the roadmap.
+
+### Option 2 Store health plan packages as JSON.
+
+This entails a single field database migration where we switch from the health plan storage format from protobuf to JSON (which can be queried in Postgres)
+### Option 3 Move health plan data out of a single table and into contract and rates tables
+
+This entails rewriting the postgres database tables for health plan package related data and performing a significant migration.
+
+## Chosen Solution: Move health plan data out of a single table and into contract and rates tables
+### Pros and Cons of the Alternatives
+​
+#### Option 1 No change 
+​
+- `+` No change to data storage
+- `±` Simplicity of mental model at first blush - we get to continue thinking of health plan data as a single standalone "package"
+- `-` Continued scope creep around features with relational rates data.
+  `-` Continued scope creep around features that seem to presume rates are easy to handle independently. 
+- `-` Shape of the data in the database does not reflect how it is used (rates seem very much to be relational). 
+- `-` The eng team would either continue emulating SQL relationships between contracts and rates in the health plan binary (as we did when we started encoding rates with manual UUIDs and referencing them across packages ) or start to duplicate data across packages and commit to maintaining those duplicates (more of a noSQL approach).
+
+#### Option 2 Store health plan packages as JSON.
+
+- `+` Simple change to data storage that would also unlock ability to query rates independently 
+- `+` We can query rates independently in postgres and deliver features like rates dashboard and rates reporting.
+- `±` Simplicity of mental model at first blush - we get to continue thinking of health plan data as a single standalone "package"
+- `-` Features that require relational rates data are still difficult.
```

Comment:
I feel a little hazy on both of the cons here.  If we can query rates independently, what's the difficult part?  And up to you whether we want to expand on the concerns about "shape" here and in the previous section.

Reply:
@rswerve That makes sense I'm trying to think how to word it- how about _"doesn't solve the issue of how to we create, edit, delete relationships between individual rates stored inside the health plan JSON and other resources in the db (e.g. between rates and contracts or between rates and questions)"_. Does that land?

Basically even if we can query with this option (yay) we can't easily track changes to the relationships because we would still be trying to relate a subset of data in the JSON to other tables or to fields on tables.

Reply:
tried to address this in [ec59b86](https://github.com/Enterprise-CMCS/managed-care-review/pull/1699/commits/ec59b86dc5a068fb1cc3611417c3cd7db1ed0b8e)